features/karaf/openhab-core/src/main/feature/feature.xml

```diff
@@ -90,6 +91,18 @@
 		<bundle>mvn:org.openhab.core.bundles/org.openhab.core.config.discovery.addon.ip/${project.version}</bundle>
 	</feature>
 
+	<feature name="openhab-core-config-discovery-sddp" version="${project.version}">
+		<feature>openhab-core-base</feature>
+		<bundle>mvn:org.openhab.core.bundles/org.openhab.core.config.discovery.sddp/${project.version}</bundle>
+	</feature>
+
+	<feature name="openhab-core-config-discovery-addon-sddp" version="${project.version}">
+		<feature>openhab-core-base</feature>
+		<feature>openhab-core-config-discovery-addon</feature>
+		<feature dependency="true">openhab-core-config-discovery-sddp</feature>
```

Comment:
You should then be able to remove this line.

Reply:
See above..

Reply:
This is actually telling you that openhab-core-config-discovery-sddp does not need to be installed when starting this feature, but it will already be there. Depending on the sequence of starting things, this may not be true I think.
See https://stackoverflow.com/questions/51481442/apache-karaf-feature-dependency

Reply:
> telling you that openhab-core-config-discovery-sddp does not need to be installed when starting this feature

Yes, I got that. However the SddpAddonFinder @activate constructor has a @reference for SddpDiscoveryService (which was in 'openhab-core-config-discovery-sddp' now moved to 'openhab-core-base') so when SddpAddonFinder is instantiated it should create a SddpDiscoveryService (if not already automatically present).

I can see in the logs that SddpDiscoveryService is indeed automatically loaded. And then when I force openhab-core-config-discovery--addon-sddp (by putting the Jar file in the addons folder, or by 'feature:install') I see the error "Not all dependencies satisfied, cannot activate". In the log below, you can see that SddpDiscoveryService is loaded and activated (automaticall) .. yet when tries to load SddpDiscoveryService says there is an unsatisfied reference to SddpDiscoveryService. Aargh @!&*%$

> Depending on the sequence of starting things.

There still seems to be a problem even when (as shown in the log time stamps) the components are started in the correct order.

```
2024-05-27 11:37:54.095 [DEBUG] [.discovery.sddp.SddpDiscoveryService] - bundle org.openhab.core.config.discovery.sddp:4.2.0.202405271007 (173)[org.openhab.core.config.discovery.sddp.SddpDiscoveryService] : BundleComponentActivator : ComponentHolder created.
..
2024-05-27 11:37:54.461 [DEBUG] [.discovery.sddp.SddpDiscoveryService] - bundle org.openhab.core.config.discovery.sddp:4.2.0.202405271007 (173)[org.openhab.core.config.discovery.sddp.SddpDiscoveryService(132)] : Changed state from satisfied to active

2024-05-27 11:38:24.519 [DEBUG] [discovery.addon.sddp.SddpAddonFinder] - bundle org.openhab.core.config.discovery.addon.sddp:4.2.0.202405271008 (269)[org.openhab.core.config.discovery.addon.sddp.SddpAddonFinder] : BundleComponentActivator : ComponentHolder created.
2024-05-27 11:38:24.525 [DEBUG] [discovery.addon.sddp.SddpAddonFinder] - bundle org.openhab.core.config.discovery.addon.sddp:4.2.0.202405271008 (269)[org.openhab.core.config.discovery.addon.sddp.SddpAddonFinder(365)] : New service tracker for $000, initial active: false, previous references: {}, classFilter: (objectClass=org.openhab.core.config.discovery.sddp.SddpDiscoveryService), initialReferenceFilter (objectClass=org.openhab.core.config.discovery.sddp.SddpDiscoveryService)
..
2024-05-27 11:38:24.532 [DEBUG] [discovery.addon.sddp.SddpAddonFinder] - bundle org.openhab.core.config.discovery.addon.sddp:4.2.0.202405271008 (269)[org.openhab.core.config.discovery.addon.sddp.SddpAddonFinder(365)] : Dependency not satisfied: $000
2024-05-27 11:38:24.532 [DEBUG] [discovery.addon.sddp.SddpAddonFinder] - bundle org.openhab.core.config.discovery.addon.sddp:4.2.0.202405271008 (269)[org.openhab.core.config.discovery.addon.sddp.SddpAddonFinder(365)] : Not all dependencies satisfied, cannot activate
```

Reply:
=> perhaps @wborn has some suggestion?